test/jdk/javax/swing/JInternalFrame/6647340/bug6647340.java

```diff
@@ -77,77 +74,69 @@ private void setupUI() {
     }
 
     private void test() throws Exception {
-        sync();
         test1();
-        sync();
+
+        robot.waitForIdle();
+        robot.delay(500);
         check1();
-        sync();
+        robot.waitForIdle();
+
         test2();
-        sync();
+        robot.waitForIdle();
+        robot.delay(500);
         check2();
     }
 
     private void test1() throws Exception {
-        SwingUtilities.invokeAndWait(new Runnable() {
-            public void run() {
-                setIcon(true);
-                location = jif.getDesktopIcon().getLocation();
-                Dimension size = frame.getSize();
-                frame.setSize(size.width + 100, size.height + 100);
-            }
+        SwingUtilities.invokeAndWait(() -> {
+            setIcon(true);
+            location = jif.getDesktopIcon().getLocation();
+            Dimension size = frame.getSize();
+            frame.setSize(size.width + 100, size.height + 100);
         });
     }
 
     private void test2() throws Exception {
-        SwingUtilities.invokeAndWait(new Runnable() {
-            public void run() {
-                setIcon(false);
-            }
+        SwingUtilities.invokeAndWait(() -> {
+            setIcon(false);
         });
-        sync();
-        SwingUtilities.invokeAndWait(new Runnable() {
-            public void run() {
-                Dimension size = frame.getSize();
-                frame.setSize(size.width - 100, size.height - 100);
-            }
+        robot.waitForIdle();
+        robot.delay(500);
+
+        SwingUtilities.invokeAndWait(() -> {
+            Dimension size = frame.getSize();
+            frame.setSize(size.width - 100, size.height - 100);
         });
-        sync();
-        SwingUtilities.invokeAndWait(new Runnable() {
-            public void run() {
-                setIcon(true);
-            }
+        robot.waitForIdle();
+        robot.delay(500);
+
+        SwingUtilities.invokeAndWait(() -> {
+            setIcon(true);
         });
     }
 
-    private void check1() {
-        if (!jif.getDesktopIcon().getLocation().equals(location)) {
+    private void check1() throws Exception {
+        SwingUtilities.invokeAndWait(() -> {
+            iconloc = jif.getDesktopIcon().getLocation();
+        });
+        if (!iconloc.equals(location)) {
             System.out.println("First test passed");
         } else {
             throw new RuntimeException("Icon isn't shifted with the frame bounds");
         }
     }
 
-    private void check2() {
-        if (jif.getDesktopIcon().getLocation().equals(location)) {
+    private void check2() throws Exception {
+        SwingUtilities.invokeAndWait(() -> {
+            iconloc = jif.getDesktopIcon().getLocation();
+        });
+        if (iconloc.equals(location)) {
```

Comment:
iconloc and iconloc are set on EDT and used on the main thread, it would be good to mark them volatile.